libs/geo/src/main/java/org/elasticsearch/geo/geometry/Rectangle.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.geo.geometry;
+
+/**
+ * Represents a lat/lon rectangle in decimal degrees.
+ */
+public class Rectangle implements Geometry {
+    public static final Rectangle EMPTY = new Rectangle();
+    /**
+     * maximum longitude value (in degrees)
+     */
+    private final double minLat;
+    /**
+     * minimum longitude value (in degrees)
+     */
+    private final double minLon;
+    /**
+     * maximum latitude value (in degrees)
+     */
+    private final double maxLat;
+    /**
+     * minimum latitude value (in degrees)
+     */
+    private final double maxLon;
+
+    private final boolean empty;
+
+    private Rectangle() {
+        minLat = 0;
+        minLon = 0;
+        maxLat = 0;
+        maxLon = 0;
+        empty = true;
+    }
+
+    /**
+     * Constructs a bounding box by first validating the provided latitude and longitude coordinates
+     */
+    public Rectangle(double minLat, double maxLat, double minLon, double maxLon) {
+        GeometryUtils.checkLatitude(minLat);
+        GeometryUtils.checkLatitude(maxLat);
+        GeometryUtils.checkLongitude(minLon);
+        GeometryUtils.checkLongitude(maxLon);
+        this.minLon = minLon;
+        this.maxLon = maxLon;
+        this.minLat = minLat;
+        this.maxLat = maxLat;
+        empty = false;
+        if (maxLat < minLat) {
+            throw new IllegalArgumentException("max lat cannot be less than min lat");
```

Comment:
shouldn't be a check for minLon maxLon as well?

Reply:
Nope, it is perfectly valid to have minLon > maxLon, it just means that the Rectangle crosses the dateline.